src/RestControllers/FHIR/FhirPatientRestController.php

```diff
@@ -58,14 +58,22 @@ public function put($pid, $fhirJson)
         }
         $data = $this->fhirPatientService->parsePatientResource($fhirJson);
 
-        $validationResult = $this->fhirPatientService->validate($data);
+        if ($this->fhirPatientService->isPatient($pid)) {
+            $validationResult = $this->fhirPatientService->validate($data, 'optional');
+            $action = 'update';
+            $status = 200;
+        } else {
+            $validationResult = $this->fhirPatientService->validate($data);
+            $action = 'insertById';
+            $status = 201;
+        }
```

Comment:
this stuff seems odd. shouldn't POST always be insert and PUT always be update?

Reply:
what if user adds a data with an id which doesn't exist in the system in such case as per FHIR guide they suggest to create one and give response with status 201 

https://www.hl7.org/fhir/http.html#update
they are using "MAY" here so we can ignore this ?

Reply:
That seem a bit odd to offer though. For example, what if they do:
pid
mname

Just to change the mname. And now if the pid does not exist yet, OpenEMR will add:
1. A user with just a mname set
2. And directly setting the pid could be very problematic and break things.

Reply:
IMO this should be the job of patch and update will update the whole resource and if not found will create one.

PATCH will just patch some data of the existing user and never create a new resource.

Reply:
![image](https://user-images.githubusercontent.com/47683556/78420910-302f2700-7671-11ea-928a-455ab7b95df7.png)

![image](https://user-images.githubusercontent.com/47683556/78420912-391ff880-7671-11ea-9a1e-678c94d7ca82.png)

Reply:
I honestly didn't even know about the PUT/PATCH stuff. Just did a nice read here:
https://medium.com/backticks-tildes/restful-api-design-put-vs-patch-4a061aa3ed0b

If we wish to support PUT/PATCH correctly, looks like we'd need to change calls/routes a bit (I would not be against this and would defer to @sjpadgett and others such as yourself with REST expertise on this). We would have a tough time supporting some PUT calls (like for patient) since setting whatever pid you want can really break things. Of course when we migrate to uuid rather than id in tables, then this kind of thing will be much easier (in the super far distant future).

Reply:
A good first step will be adding PATCH here, so it even works on the dockers:
https://github.com/openemr/openemr-devops/blob/master/docker/openemr/5.0.3/openemr.conf#L25

Reply:
We can leave that create new user part from PUT for now as it is an optional thing to do.

Reply:
> what if user adds a data with an id which doesn't exist in the system in such case as per FHIR guide they suggest to create one and give response with status 201
> 
> https://www.hl7.org/fhir/http.html#update
> they are using "MAY" here so we can ignore this ?

Hello @yashrajbothra with the current API's if one tries an update with an ID that doesn't exist, usually nothing is returned, and no update is done on the database. I agree it would be a nice to have to create a better response, which could work with a method that sees if IDs used across API's are valid and already existing. I do think you can break your Current PR into smaller PR's and of course create issues for each PR first.

Reply:
The majority of REST apis I've worked with rarely implement PUT/PATCH as the resources involved over GET/POST are more complicated and difficult to implement. 

I've never seen any REST implementation that lets PUT create a resource even though the RFC specifies it.  Just my 2 cents.  Creation of resources is supposed to be done via POST.